src/Michał/Fsd.Michal.Cs.Ex2/Shop/Shop/Sell.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+
+namespace Shop
+{
+    public class Sell
+    {
+        public Warehouse warehouse { get; set; }
+
+        Product product { get; set; }
+
+        public void CustomerService(Warehouse warehouse)
+        {
+            string input;
+
+            do
+            {
+                ShowMenuOptions();
+                input = Console.ReadLine();
+
+                switch (input)
+                {
+                    case "1":
+                        warehouse.BuyProduct();
+                        Console.WriteLine();
+                        DoYouWantToLeaveShop();
+                        break;
+
+                    case "2":
+                        Console.Clear();
+                        warehouse.ShowProducts();
+                        Console.WriteLine();
+                        DoYouWantToLeaveShop();
+                        break;
+
+                    case "3":
+                        warehouse.AddProduct(product);
+                        Console.WriteLine();
+                        DoYouWantToLeaveShop();
+                        Console.Clear();
+                        break;
+
+                    case "4":
+                        DoYouWantToLeaveShop();
+                        break;
+
+                    default:
+                        Console.Clear();
+                        Console.WriteLine("incorrect key - try again");
+                        DoYouWantToLeaveShop();
+                        break;
+
+                }
+            } while (true);
+        }
+
+        internal void ShowMenuOptions()
```

Comment:
klasa "Sell" ma przynajmniej 2 odpowiedzialności - sposób prezentowania i logikę zachowania sklepu. Jeżeli chcesz (o ile dobrze myślę) zrobić z niej klasę I/O pytającą usera o zdanie, to nie trzymaj w niej instancji Product i Warehouse. Odseparuj to żeby było zgodne z SOLID